### PR TITLE
Linux build

### LINKED ISSUE #13: Windows Installer
Please add a Windows installer.

- [x] Generated by the `npm run release` script

### LINKED ISSUE #17: Linux Build Release
## Environment
Linux Mint 18.3

## Bug
Cannot build app on Linux computer, `npm run release` returns the following error message shown below. The full error output is attached.
[terminal_output.txt](https://github.com/QutEcoacoustics/ap-desktop/files/2924174/terminal_output.txt)
[2019-03-04T02_54_22_793Z-debug.log](https://github.com/QutEcoacoustics/ap-desktop/files/2924175/2019-03-04T02_54_22_793Z-debug.log)

```
An unhandled error has occurred inside Forge:
An error occured while making for target: deb
Error creating contents of package: Error creating binary symlink: could not find the Electron app binary at "/home/charles/Documents/GitHub/ap-desktop/out/AP Desktop-linux-x64/ap_desktop". You may need to re-bundle the app using Electron Packager's "executableName" option.
Error: Error creating contents of package: Error creating binary symlink: could not find the Electron app binary at "/home/charles/Documents/GitHub/ap-desktop/out/AP Desktop-linux-x64/ap_desktop". You may need to re-bundle the app using Electron Packager's "executableName" option.
    at fs.pathExists.then.exists (/home/charles/Documents/GitHub/ap-desktop/node_modules/electron-installer-common/src/installer.js:142:17)
```

## Expected Behaviour
Compiling system on linux should build a production version for all major linux systems if possible.

### PR DESCRIPTION
**Pull request by Allcharles**
Added ability for `npm run make` command to build both linux (.deb) and windows (.exe) files and installers. Program does not yet have icons, but they have been added to a res folder. This also contains the basics of a Travis CI file for automated testing and compilation.
Fixes #17 
Fixes #13 (uses `npm run make` to generate programs, this is the default name for electron but can be changed later)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  - sh -e /etc/init.d/xvfb start
+  - npm install
+
+script:
+  - npm test
+  - npm run make
+
+env:
+  - dist: xenial
+  - os: windows
```
Comment (atruskie) [inline .travis.yml]: what is this `os` env variable about?

File: README.md [hunk 2]
```diff
@@ -15,6 +15,10 @@ AP Desktop is a Graphical User Interface (GUI) for the open source program Analy
 - To build the project, run the following command: `npm run release`
 - You will need to run this once on each patform (Windows/Linux/Mac) to get the full suit
 
+### Linux
+
+- To package the application you will need to install wine [link](https://wiki.winehq.org/Download)
```
Comment (atruskie) [inline README.md]: what is wine needed for?

File: package.json [hunk 3]
```diff
@@ -1,24 +1,25 @@
...
-    "babel": "npx babel ./src/js/ --out-dir ./src/js-compiled/ && npx babel ./tests/test/ --out-dir ./tests/test-compiled/",
-    "babel:async": "npx babel ./src/js/ --out-dir ./src/js-compiled/ -w"
+    "package": "electron-forge package",
+    "make": "npm run clean && electron-forge make",
+    "publish": "electron-forge publish",
+    "lint": "eslint src --color"
   },
   "keywords": [],
-  "author": "Charles Alleman",
+  "author": "Queensland University of Technology",
```
Comment (atruskie) [inline package.json]:
```suggestion
  "author": "QUT Ecoacoustics",
```

You should add yourself to the contributors field

File: package.json [hunk 4]
```diff
@@ -30,45 +31,60 @@
...
           "deb"
         ]
       },
       "electronPackagerConfig": {
         "packageManager": "npm"
       },
       "electronWinstallerConfig": {
-        "name": "ap_desktop"
+        "setupExe": "setup.exe",
+        "authors": "Queensland University of Technology"
```
Comment (atruskie) [inline package.json]:
```suggestion
        "authors": "QUT Ecoacoustics"
```

File: res/qut logo.svg [hunk 5]
```diff
@@ -0,0 +1,48 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (atruskie) [inline res/qut logo.svg]: Please remove this file. It violates brand guidelines, and further, we need written permission to customize it.

File: src/index.js [hunk 6]
```diff
@@ -17,7 +18,8 @@ const createWindow = () => {
   mainWindow = new BrowserWindow({
     width: 800,
     height: 600,
-    minWidth: 650
+    minWidth: 650,
+    icon: path.join(__dirname, "res/qut-logo-64.png")
```
Comment (atruskie) [inline src/index.js]: fix when removing logo

**Review by atruskie — CHANGES_REQUESTED** [review]
The Travis build was failing, but I see you've fixed it in the meantime 👍

**Comment by atruskie** [comment]
Also, you'll need to get the build to actually pass! https://travis-ci.org/QutEcoacoustics/ap-desktop

**Comment by Allcharles** [comment]
Still has the issue with the system downloading AP resulting in a API key overload. I will fix later

**Review by atruskie — COMMENTED** [review]
this token needs to be encrypted in the travis yaml and referenced here as an environment variable

**Review by atruskie — APPROVED** [review]

**Pull request merged**